quill-jdbc/src/main/scala/io/getquill/context/jdbc/JdbcDecoders.scala

```diff
@@ -20,10 +20,10 @@ trait JdbcDecoders {
   implicit def optionDecoder[T](implicit d: Decoder[T]): Decoder[Option[T]] =
     new Decoder[Option[T]] {
       def apply(index: Int, row: ResultSet) = {
-        val res = d(index, row)
+        // don't invoke the parent decoder before the value was null checked
         row.wasNull match {
```

Comment:
of course.. I was just a little bit dumb about it. any idea's how to make it better? I.e. maybe catch a null pointer exception?